stacker/config/__init__.py

```diff
@@ -87,12 +93,18 @@ def render(raw_config, environment=None):
     if not environment:
         environment = {}
     try:
-        buff.write(t.substitute(environment))
-    except KeyError, e:
+        substituted = t.substitute(environment)
```

Comment:
I think we can clean this up some:

```
try:
    substituted = t.substitute(environment)
except KeyError as e:
    raise exceptions.MissingEnvironment(e.args[0])
except ValueError:
    substituted = t.safe_substitute(environment)

if not isinstance(substituted, str):
    substituted = substituted.decode("utf-8")

buff.write(substituted)
```

Reply:
I'll see what I can do.

Reply:
Yeah, you are right, this will do.

However, when I was reading this part one more time, I realized that there could be another issue here. This try-except block might not always work as what we had expected it to do. substitute will always run sequentially, and it will panic at the first sight of error. So if we have a config file/environment values pair, that potentially will throw a `KeyError` and a `ValueError`, but the `KeyError` will only occur after the `ValueError`, then `KeyError` will not be raised.

Anyway, this is just a side note, and I think it does not fit into this PR in any way.